src/library/Uno.Material/MaterialColors.cs

```diff
@@ -28,18 +28,35 @@ public string OverrideSource
 		private static void OnColorPaletteOverrideSourceChanged(DependencyObject dependencyObject, DependencyPropertyChangedEventArgs args)
 		{
 			ColorPaletteOverrideSource = args.NewValue as string;
+
+			ApplyOverride();
 		}
 
-		public MaterialColors()
+		internal static void ApplyOverride(ResourceDictionary target = null)
 		{
-			MergedDictionaries.Add(new ResourceDictionary { Source = new Uri("ms-appx:///Uno.Material/Styles/Application/ColorPalette.xaml") });
 			if (!string.IsNullOrWhiteSpace(ColorPaletteOverrideSource))
 			{
-				MergedDictionaries.Add(new ResourceDictionary { Source = new Uri(ColorPaletteOverrideSource) });
-			}
+				var overrideDictionary = new ResourceDictionary { Source = new Uri(ColorPaletteOverrideSource) };
+
+				target = target ?? Application.Current?.Resources;
```

Comment:
is this assuming that the resources are being added directly in App.Resources? What's going to happen for other scenarios, eg included in a nested dictionary, or included in per-Page Resources?

Reply:
This is now removed, as unused.